lib/utils/compose_ts.ts

```diff
@@ -1314,12 +1316,23 @@ export async function deployProject(
 
 	let $release: Release;
 	try {
+		let contractContent: string | undefined;
+		let contract = await tryGetContractContent(`${projectPath}/balena.yaml`);
+		if (!contract) {
+			contract = await tryGetContractContent(`${projectPath}/balena.yml`);
+		}
+		if (contract) {
```

Comment:
Is contract content allowed without a (valid) contract version? If not, it might make sense to check for a valid contract version. See other comment for contract validation.

Reply:
I believe it is ok to have no version.

Reply:
I wonder if we should validate semver as well. If an invalid semver is used, the API request will fail, but not sure what is the good practice here.